### ISSUE TITLE
Provisioning profile "match AppStore old.bundle.identifier" has app ID "old.bundle.identifier", which does not match the bundle ID "new.bundle.identifier"

### ISSUE THREAD

**Issue by alexbuicescu** [fastlane/fastlane#10244]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I have **Multiple targets of the same underlying app**, but the problem is when I'm changing the bundle identifier, the new provisioning profile from match (the one for the new bundle identifier) is not being set. For example:
1. Match ```Installed Provisioning Profile```. Everything is ok here
2. After match:
  ```
  Detected provisioning profile mapping: {"new.bundle.identifier"=>"match AppStore new.bundle.identifier", ""=>"match AppStore old.bundle.identifier"}
  ```
  Notice the **match AppStore old.bundle.identifier** which I think shouldn't be here
3. Summary for gym
```
+--------------------------------------+--------------------------------------+
|                           Summary for gym 2.53.1                            |
+--------------------------------------+--------------------------------------+
| export_method                        | app-store                            |
| export_options.provisioningProfiles  | match AppStore                       |
| .new.bundle.identifier               | new.bundle.identifier                |
| export_options.provisioningProfiles  | match AppStore                       |
| .                                    | old.bundle.identifier                |
| workspace                            | ios/instacarFront.xcworkspace        |
| scheme                               | instacarFront                        |
| export_team_id                       | our_team_id                          |
| configuration                        | Release                              |
| clean                                | true                                 |
| destination                          | generic/platform=iOS                 |
| output_name                          | instacarFront                        |
| build_path                           | /Users/alexbuicescu/Library/Develop  |
|                                      | er/Xcode/Archives/2017-09-03         |
| output_directory                     | .                                    |
| silent                               | false                                |
| skip_package_ipa                     | false                                |
| buildlog_path                        | ~/Library/Logs/gym                   |
| xcode_path                           | /Applications/Xcode.app              |
+--------------------------------------+--------------------------------------+

$ set -o pipefail && xcodebuild -workspace ios/instacarFront.xcworkspace -scheme instacarFront -configuration Release -destination 'generic/platform=iOS' -archivePath /Users/alexbuicescu/Library/Developer/Xcode/Archives/2017-09-03/instacarFront\ 2017-09-03\ 13.07.28.xcarchive clean archive | tee /Users/alexbuicescu/Library/Logs/gym/instacarFront-instacarFront.log | xcpretty
```
4. At the end of gym I get:
```
Check dependencies
Provisioning profile "match AppStore old.bundle.identifier" has app ID "old.bundle.identifier", which does not match the bundle ID "new.bundle.identifier"
Code signing is required for product type 'Application' in SDK 'iOS 10.3'
Code signing is required for product type 'Application' in SDK 'iOS 10.3'
Code signing is required for product type 'Application' in SDK 'iOS 10.3'
```

If I go to Xcode I can select the provisioning profile by hand after running match and before running gym and everything works ok.
### Environment
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.48.0"
require 'fileutils'

platform :ios do
  desc "Submit a new build to App Store"
  lane :production_final do |options|

    disable_automatic_code_signing(path: "ios/instacarFront.xcodeproj")
    match()
    gym()
    enable_automatic_code_signing(path: "ios/instacarFront.xcodeproj")

    # use deliver for app store release
    deliver(
         submit_for_review: true,
         force: true,
         automatic_release: true,
         price_tier: 0,
         submission_information: {
               add_id_info_limits_tracking: false,
               add_id_info_serves_ads: false,
               add_id_info_tracks_action: true,
               add_id_info_tracks_install: true,
               add_id_info_uses_idfa: false,
               content_rights_has_rights: false,
               content_rights_contains_third_party_content: false,
               export_compliance_platform: 'ios',
               export_compliance_compliance_required: false,
               export_compliance_encryption_updated: false,
               export_compliance_app_type: nil,
               export_compliance_uses_encryption: false,
               export_compliance_is_exempt: false,
               export_compliance_contains_third_party_cryptography: false,
               export_compliance_contains_proprietary_cryptography: false,
               export_compliance_available_on_french_store: false
             }
    )
  end

end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file "deploy/google-play-api-secret.json"
package_name ENV["package_name"]

app_identifier ENV["bundle_identifier"]       # The bundle identifier of your app
username "our@apple.id" # your Apple ID user
apple_id "our@apple.id" # your Apple ID user

itunes_connect_id "our@apple.id"     # iTunes Connect Account

team_id ENV["ios_team"] # Developer Portal Team ID
```
</details>

<details><summary>`./fastlane/Matchfile`</summary>

```ruby
git_url "our_git_url"
type "appstore" # The default type, can be: appstore, adhoc, enterprise or development

app_identifier ENV["bundle_identifier"]
username "our@apple.id" # Your Apple Developer Portal username
git_branch ENV["match_file_git_branch"]
team_id ENV["itunes_connect_team_id"] # Developer Portal Team ID
```
</details>

<details><summary>`./fastlane/Gymfile`</summary>

```ruby
export_method "app-store"  # Method used to export the archive. Valid values are: app-store, ad-hoc, package, enterprise, development, developer-id
workspace "ios/instacarFront.xcworkspace"
scheme "instacarFront"
export_team_id ENV["itunes_connect_team_id"]
configuration "Release"
clean true
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.55.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
</details>

</details>

**Comment by KrauseFx** [comment]
This is related to your project setup, make sure to follow the code signing guide on how to setup code signing in your Xcode project 👍

**Comment by alexbuicescu** [comment]
Looks like my issue might be a duplicate of https://github.com/fastlane/fastlane/issues/8973. @col workaround worked for me.

**Comment by col** [comment]
If you’re using the workaround in [#8973](https://github.com/fastlane/fastlane/issues/8973) it’s no longer required. I ended up fixing the core issue in update_provisioning_profile and it was released in fastlane 2.56.0. No need to use xcargs. 👍
